jung-visualization/src/main/java/edu/uci/ics/jung/visualization/layout/AggregateLayoutModel.java

```diff
@@ -215,33 +199,30 @@ public LayoutStateChangeSupport getLayoutStateChangeSupport() {
    *
    * @return the location of the node
    */
-  public P apply(N node) {
-    PointModel<P> pointModel = delegate.getPointModel();
-    for (LayoutModel<N, P> layoutModel : layouts.keySet()) {
+  public Point apply(N node) {
+    for (LayoutModel<N> layoutModel : layouts.keySet()) {
       if (layoutModel.getGraph().nodes().contains(node)) {
-        P center = layouts.get(layoutModel);
+        Point center = layouts.get(layoutModel);
         // transform by the layout itself, but offset to the
         // center of the sublayout
         int width = layoutModel.getWidth();
         int height = layoutModel.getHeight();
         AffineTransform at =
-            AffineTransform.getTranslateInstance(
-                pointModel.getX(center) - width / 2, pointModel.getY(center) - height / 2);
-        P nodeCenter = layoutModel.apply(node);
+            AffineTransform.getTranslateInstance(center.x - width / 2, center.y - height / 2);
+        Point nodeCenter = layoutModel.apply(node);
         log.trace("sublayout center is {}", nodeCenter);
-        double[] srcPoints =
-            new double[] {pointModel.getX(nodeCenter), pointModel.getY(nodeCenter)};
+        double[] srcPoints = new double[] {nodeCenter.x, nodeCenter.y};
         double[] destPoints = new double[2];
         at.transform(srcPoints, 0, destPoints, 0, 1);
-        return pointModel.newPoint(destPoints[0], destPoints[1]);
+        return new Point(destPoints[0], destPoints[1]);
       }
     }
     return delegate.apply(node);
   }
 
   /** @return {@code true} iff the delegate layout and all sublayouts are done */
   public boolean done() {
-    for (LayoutModel<N, P> layoutModel : layouts.keySet()) {
+    for (LayoutModel<N> layoutModel : layouts.keySet()) {
       if (layoutModel instanceof IterativeContext) {
```

Comment:
I don't spot any implementations of `LayoutModel` that implement `IterativeContext`; that seems to apply to the algorithm, not to the model.  I also don't see any way of querying the `LayoutModel` for the `LayoutAlgorithm` that it's using (because IIUC it only uses it during `accept()` and methods it calls) so AFAICT this method would just always return true.  Am I missing something, or is there a bug here (and possibly a structural issue)?

Reply:
that has to be a bug left over from when Layouts were also LayoutModels.

Reply:
Makes sense.  I assume that this means that we can just excise that entire method (which itself would only make sense if `LayoutModel` were an `IterativeContext`).